app/src/main/java/com/asfoundation/wallet/ui/iab/MergedAppcoinsFragment.kt

```diff
@@ -0,0 +1,346 @@
+  private var onBackPressSubject: PublishSubject<Any>? = null
+  private lateinit var iabView: IabView
+
+  private val fiatAmount: BigDecimal by lazy {
+    if (arguments!!.containsKey(FIAT_AMOUNT_KEY)) {
+      arguments!!.getSerializable(FIAT_AMOUNT_KEY) as BigDecimal
+    } else {
+      throw IllegalArgumentException("amount data not found")
+    }
+  }
+  private val currency: String by lazy {
+    if (arguments!!.containsKey(FIAT_CURRENCY_KEY)) {
+      arguments!!.getString(FIAT_CURRENCY_KEY)
+    } else {
+      throw IllegalArgumentException("currency data not found")
+    }
+  }
+
+  private val bonus: String by lazy {
+    if (arguments!!.containsKey(BONUS_KEY)) {
+      arguments!!.getString(BONUS_KEY)
+    } else {
+      throw IllegalArgumentException("bonus data not found")
+    }
+  }
+
+  private val appName: String by lazy {
+    if (arguments!!.containsKey(APP_NAME_KEY)) {
+      arguments!!.getString(APP_NAME_KEY)
+    } else {
+      throw IllegalArgumentException("app name data not found")
+    }
+  }
+
+  private val productName: String? by lazy {
+    if (arguments!!.containsKey(PRODUCT_NAME_KEY)) {
+      arguments!!.getString(PRODUCT_NAME_KEY)
+    } else {
+      throw IllegalArgumentException("product name data not found")
+    }
+  }
+
+  private val appcAmount: BigDecimal by lazy {
+    if (arguments!!.containsKey(APPC_AMOUNT_KEY)) {
+      arguments!!.getSerializable(APPC_AMOUNT_KEY) as BigDecimal
+    } else {
+      throw IllegalArgumentException("appc data not found")
+    }
+  }
+
+  private val appcEnabled: Boolean by lazy {
+    if (arguments!!.containsKey(APPC_ENABLED_KEY)) {
+      arguments!!.getBoolean(APPC_ENABLED_KEY)
+    } else {
+      throw IllegalArgumentException("appc enable data not found")
+    }
+  }
+
+  private val creditsEnabled: Boolean by lazy {
+    if (arguments!!.containsKey(CREDITS_ENABLED_KEY)) {
+      arguments!!.getBoolean(CREDITS_ENABLED_KEY)
+    } else {
+      throw IllegalArgumentException("credits enable data not found")
+    }
+  }
+
+  private val isBds: Boolean by lazy {
+    if (arguments!!.containsKey(IS_BDS_KEY)) {
+      arguments!!.getBoolean(IS_BDS_KEY)
+    } else {
+      throw IllegalArgumentException("is bds data not found")
+    }
+  }
+
+  private val isDonation: Boolean by lazy {
+    if (arguments!!.containsKey(IS_DONATION_KEY)) {
+      arguments!!.getBoolean(IS_DONATION_KEY)
+    } else {
+      throw IllegalArgumentException("is donation data not found")
+    }
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    paymentSelectionSubject = PublishSubject.create()
+    onBackPressSubject = PublishSubject.create()
+    mergedAppcoinsPresenter = MergedAppcoinsPresenter(this, CompositeDisposable())
+  }
+
+  override fun onAttach(context: Context) {
+    super.onAttach(context)
+    if (context !is IabView) {
+      throw IllegalStateException("Regular buy fragment must be attached to IAB activity")
```

Comment:
This should refer to the correct fragment name.
Plus: change all ocurrences of "Regular buy fragment" to the correct one 😄